---
title: "Depth-dependent genetic structuring of *Montastraea cavernosa* and their Symbiodiniaceae communities varies between Alacranes and Bajos del Norte reefs, Campeche Bank, Mexico"
author: "Alexis Sturm"
date: "8/31/2021"
output: 
html_document:
    theme: flatly
    toc: yes
    toc_depth: 4
    toc_float: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, fig.align = 'left')
#knitr::opts_knit$set(root.dir = '../revisedData')
options(width = 88, scipen = 4)
```

***

# About this document
***  
All analyses were performed in R version 3.6.2.
This is the code that accompanies the publication *Depth-dependent genetic structuring of Montastraea cavernosa and their Symbiodiniaceae communities varies between Alacranes and Bajos del Norte reefs, Campeche Bank, Mexico*. Here you will find all the code to repeat the statistical analyses performed in R and the figures created for the manuscript. The accompanying library preparation protocol and bioinformatic walkthrough can be found by clicking on the links.

If you have any issues or questions about the code please feel free to send an email to lexie.sturm@gmail.com.

[![DOI](https://zenodo.org/badge/381362197.svg)](https://zenodo.org/badge/latestdoi/381362197)

***

# Basic setup of R environment
***
## Loading required packages
Make sure to set the working directory:
`setwd("~/path/to/directory/with/data")`

```{r, set working directory}
setwd('/Users/student/Documents/GitHub/mxMcavSnp/')
```

For the following analyses we will require the use of multiple different R packages, we can use the package *pacman* to quickly load them.
```{r, load packages, include = TRUE, message = FALSE, warning = FALSE, results = 'hide'}
if (!require("pacman")) install.packages("pacman")
pacman::p_load("adegenet", "dendextend", "flextable", "gdata", "ggdendro", "hierfstat", "Imap", "kableExtra", "paletteer", "patchwork", "officer", "poppr", "RColorBrewer", "reshape2", "StAMPP", 
"tidyverse", "vcfR", "vegan", "WGCNA", "ggnewscale", "rnaturalearth", "sf", "ggspatial", 
"cowplot", "marmap", "dplyr", "edgeR", "MCMC.OTU", "pairwiseAdonis", "distGeo", "pcadapt")

pacman::p_load_gh("eliocamp/ggnewscale", "ropensci/rnaturalearthhires")
```

# Figures and analyses
##Sampling Map
```{r, sampling map}
mxSites = read.csv("mxSampleSites.csv", header = TRUE)
mxSites$depthZone=as.factor(mxSites$depthZone)
# coerce to sf object

mxSites1 <- st_as_sf(x = mxSites, coords = c('lon', 'lat'), crs = 4326)
mxSitesDist=st_distance(mxSites1)
mxSitesDistKm=mxSitesDist/1000
rownames(mxSitesDistKm) <-c(mxSites$reef)
colnames(mxSitesDistKm) <-c(mxSites$reef)
#mxSites$depth = factor(mxSites$depth, levels = levels(mxSites$depth)[c(2,1)])
#mxSites$reef = factor(mxSites$reef, levels = levels(mxSites$reef)[c(1, 7, 8, 3, 6, 2, 5, 4, 11, 10, 9, 12)])
mxReefs = mxSites %>% group_by(reef) %>% summarize(lat = mean(lat), lon = mean(lon))
#mxReefs$depthShape=c("Mesophotic", "Shallow", "Shallow", "Shallow", "Shallow", "Shallow", "Shallow", "Shallow", "Paired", "Paired", "Paired", "Paired")
#mxReefs$depthShape=as.factor(mxReefs$depthShape)
#mxReefs$depthShape = factor(mxReefs$depthShape, levels = levels(mxReefs$depthShape)[c(2,3,1)])
gomCountries = st_as_sf(ne_countries(country = c("United States of America", "Cuba", "Mexico", "The Bahamas", "Belize", "Guatemala", "Cayman Islands"), scale = "Large"))


#gom5.100Bathy=st_read("/Users/student/Documents/GitHub/mxMcavSnp/gom_5-100m/w98e78n31s18_isobath_5-100m.shp") %>% st_transform(54030)
gom200Bathy=st_read("/Users/student/Documents/GitHub/mxMcavSnp/GOM_isobath_100-1000m/w98e78n31s18_isobath_100-1000m.shp") %>%
filter(CONTOUR == "-200") %>% st_transform(54030)

mxPal=c("seagreen3", "blue")

mxMapA = ggplot() +
  geom_sf(data = gomCountries, fill = "white", size = 0.25) +
  geom_sf(data = gom200Bathy, aes(color = "red")) +
  geom_point(data = mxSites, aes(x = lon, y = lat, shape = depthZone), size = 0) +
  geom_point(data = mxReefs, aes(x = lon, y = lat, fill = reef), size = 3, shape=21) +
  scale_fill_manual(values = mxPal, name = "Site") +
  scale_shape_manual(values = c(24,22,23,25), breaks=c("10","15", "25", "35"), name = "Depth Zone", labels=c("10 m","15 m", "25 m", "35 m")) +
  scale_color_identity(guide="legend", labels="200 m isobath", name=NULL) +
  coord_sf(xlim = c(-103, -80), ylim = c(18.5, 30)) +
  scale_x_continuous(breaks = c(seq(-103, -80, by = 2))) +
  scale_y_continuous(breaks = c(seq(18.5, 30, by = 2))) +
  #geom_rect(aes(xmin = -89.8, xmax = -89.6, ymin = 22.35, ymax = 22.6), color = "black", fill = NA, size = 0.4) +
  #geom_rect(aes(xmin = -88.74, xmax = -88.68, ymin = 23.24, ymax = 23.32), color = "black", fill = NA, size = 0.4) +
  annotation_scale(location = "bl") +
  annotation_north_arrow(location = "tr", which_north = "true", style = north_arrow_minimal()) +
  guides(fill = guide_legend(override.aes = list(shape = 22, color = NA, size = 4), ncol = 2, order = 1), shape = guide_legend(override.aes = list(size = 3), order = 2)) +
  #guides(fill = guide_legend(override.aes = list(shape = 21, size = 4), order = 1))+ #,
         #shape = guide_legend(order=2),
         #color = guide_legend(title=NULL, order = 3))+ #override.aes = list(fill=NA),
  theme_bw() +
  theme(plot.title = element_text(size = 9) ,
        panel.background = element_rect(fill = "aliceblue"),
        plot.background = element_blank(),
        axis.title = element_blank(),
        legend.position = "bottom")

#mxMapA

#ggsave("mxMapNoInset.png", plot = mxMapA, width = 30, height = 20, units = "cm", dpi = 600)
set.seed(300)
inset = ggplot() +
  geom_segment(aes(x = -103, xend = -80, y = 18.5, yend = 30), color = "gray92", size = .55) +
  geom_sf(data = gomCountries, fill = "white", size = 0.25) +
  geom_point(data = subset(mxSites, subset = mxSites$reef == "Alacranes"), aes(x = lon, y = lat, fill = reef, shape=depthZone), size = 3, position=position_jitter(width=0.01, height=0.01)) +
  geom_point(data = subset(mxSites, subset = mxSites$reef == "Bajos del Norte"), aes(x = lon, y = lat, fill = reef, shape=depthZone), size = 3, position=position_jitter(width=0.002, height=0.002)) +
  scale_fill_manual(values = mxPal, name = "Reef") +
  scale_shape_manual(values = c(24,22,23,25), breaks=c("10","15", "25", "35"), name = "Depth Zone", labels=c("10 m","15 m", "25 m", "35 m")) +
  #geom_jitter()+
  #geom_sf(data = gom5.100Bathy, color = "gray") +
  guides(fill = guide_legend(override.aes = list(shape = 21, color = NA, size = 4))) +
  theme_bw() +
  theme(legend.title = element_text(size = 9, face = "bold"),
        axis.title = element_blank(),
        panel.background = element_rect(fill = "aliceblue"),
        legend.text = element_text(size = 9),
        legend.position = "none",
        legend.direction = "vertical",
        legend.box = "vertical",
        panel.grid=element_blank(),
        rect = element_blank())

# inset

alacranes = inset+
  ggtitle("Alacranes") +
  # geom_sf(data = gomCountries, fill = "white", size = 0.25) +
  annotation_scale(location = "bl") +
  coord_sf(xlim = c(-89.85, -89.6), ylim = c(22.35, 22.6)) +
  scale_x_continuous(breaks = c(seq(-89.85, -89.6, by = .1))) +
  scale_y_continuous(breaks = c(seq(22.35, 22.6, by = .1)))+
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text = element_blank(),
        axis.ticks = element_blank(),
        plot.background=element_blank(),
        panel.border = element_rect(size = 1))

# alacranes

bajosDelNorte = inset +
  ggtitle("Bajos del Norte") +
  annotation_scale(location = "bl") +
  coord_sf(xlim = c(-88.74, -88.68), ylim = c(23.24, 23.32)) +
  scale_x_continuous(breaks = c(seq(-88.74, -88.68, by = .1))) +
  scale_y_continuous(breaks = c(seq(23.24, 23.32, by = .1)))+
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text = element_blank(),
        axis.ticks = element_blank(),
        plot.background=element_blank(),
        plot.title = element_text(vjust=-6.5, hjust=0.5),
        panel.border = element_rect(size = 1))

#bajosDelNorte

mxMap = ggdraw() +
  draw_plot(mxMapA) +
  #draw_plot(inset) +
  draw_plot(alacranes, x = 0.01, y = 0.25, width = 0.3, height = 0.3) +
  draw_plot(bajosDelNorte, x = 0.67, y = 0.25, width = 0.3, height = 0.3)


mxMap

ggsave("./figures/figure1.tiff", plot = mxMap, width = 25, height = 20, units = "cm", dpi = 300)
```
##Heterozygosity
```{r, Heterozygosity}
hetero=read.csv("heterozygosity.csv")
snpI2P = read.csv("mxInds2PopsNoClones.csv") 
hetero=bind_cols(snpI2P, hetero)

heteroMeans=hetero %>%
  group_by(popDepth) %>%
  summarise(meanAllSites = mean(allSites), meanVariantSites = mean(variantSites), n = n())

heteroMeans1=as.table(heteroMeans)

max(heteroStatAllSites$mean, na.rm = TRUE)
min(heteroStatAllSites$mean, na.rm = TRUE)


heteroPlot <- ggplot(hetero, aes(x=popDepth, y=variantSites)) +
  geom_boxplot()+
  theme_bw()
heteroPlot
```
##Host PCoA with IBS
```{r, Host PCoA with IBS}
snpMa = as.matrix(read.table("mxMcavNoClones.ibsMat"))
mxMds = cmdscale(snpMa, eig = TRUE, x.ret = TRUE)

# Determine percent variation captured on each axis
# Calculate the eigenvalues so later we can figure out % variation shown on each Principal Coordinate
mxSnpPcoaVar = round(mxMds$eig/sum(mxMds$eig)*100, 1)
mxSnpPcoaVar

# Format data to plot
mxSnpPcoaValues = mxMds$points
mxSnpPcoaValues

snpI2P = read.csv("mxInds2PopsNoClones.csv") # 2-column tab-delimited table of individual assignments to populations; must be in the same order as samples in the bam list or vcf file.
row.names(snpI2P) = snpI2P[,1]
mxSnpPcoaValues=cbind(snpI2P, mxSnpPcoaValues)
mxSnpPcoaValues =as.data.frame(mxSnpPcoaValues, sample = rownames(mxSnpPcoaValues))
colnames(mxSnpPcoaValues)[5] <- "PCo1"
colnames(mxSnpPcoaValues)[6] <- "PCo2"
mxSnpPcoaValues

snpPCoA = merge(mxSnpPcoaValues, aggregate(cbind(mean.x=PCo1,mean.y=PCo2)~popDepth, mxSnpPcoaValues, mean), by="popDepth")

snpPCoA$depthZoneM=as.factor(snpPCoA$depthZoneM)
#snpPCoA$depthZone = factor(snpPCoA$depthZone, levels(snpPCoA$depthZone)[c(2,1)])

# SNP PCoA biplot
mxSnpPcoaPlotA = ggplot(snpPCoA, aes(x = PCo1, y = PCo2, color = site, fill = site, shape = depthZoneM, linetype = depthZoneM)) +
  geom_hline(yintercept = 0, color = "gray90", size = 0.5) +
  geom_vline(xintercept = 0, color = "gray90", size = 0.5) +
  #stat_ellipse(data = subset(snpPCoA, type = "n", geom = "polygon")) + #ellipse
  #scale_linetype_manual(values=c(1,2,4,3), name = "Depth Zone")+
  geom_point(aes(x = PCo1, y = PCo2, shape = depthZoneM), size = 3, alpha = 0.3, show.legend = FALSE, guide=FALSE) + #individual's points indicated by circles
  scale_shape_manual(values = c(24,22,23,25), breaks=c("10","15", "25", "35"), name = "Depth Zone", labels=c("10 m","15 m", "25 m", "35 m")) +
  geom_point(aes(x = mean.x, y = mean.y, shape = depthZoneM), size = 5, color = "black") + #population centroids indicated by triangles
  #scale_fill_paletteer_d("LaCroixColoR::PassionFruit", breaks=c("Alacranes", "Bajos del Norte"), name= "Population")+
  scale_fill_manual(values = c("seagreen3", "blue"), name= "Population")+
  #scale_color_paletteer_d("LaCroixColoR::PassionFruit", breaks=c("Alacranes", "Bajos del Norte"), name= "Population", guide=FALSE) +
  scale_color_manual(values = c("seagreen3", "blue"), name= "Population")+
  xlab(paste ("PCo 1 (", mxSnpPcoaVar[1],"%)", sep = "")) + #Prints percent variation explained by first axis
  ylab(paste ("PCo 2 (", mxSnpPcoaVar[2],"%)", sep = "")) + #Prints percent variation explained by second axis
  guides(shape = guide_legend(order = 2), fill = guide_legend(override.aes = list(shape = 22, size = 4, color = NA), order = 1))+ #linetype = guide_legend(order = 3),
  theme_bw()

mxSnpPcoaPlot = mxSnpPcoaPlotA +
  theme(axis.title.x = element_text(color = "black", size = 10),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.line.x = element_blank(),
        axis.title.y = element_text(color = "black", size = 10),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.line.y = element_blank(),
        legend.position = "left",
        panel.border = element_rect(color = "black", size = 1.2),
        panel.background = element_rect(fill = "white"),
        plot.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())



mxSnpPcoaPlot

ggsave("./figures/figure2.tiff", plot = mxSnpPcoaPlot, height = 5, width = 7, units = "in", dpi = 300)
```

## AMOVA and Pairwise Fst
```{r, AMOVA and Pairwise Fst, fig.width = 13.4, fig.height = 5.9}
mxVcf = read.vcfR("mxMcavNoClones.bcf")
mxGenlightPopDepth = vcfR2genlight(mxVcf, n.cores = 2) # Converts the vcf file into a file format that poppr uses the "genlight" format
locNames(mxGenlightPopDepth) = paste(mxVcf@fix[,1],mxVcf@fix[,2],sep="_")
popData = read.csv("mxInds2PopsNoClones.csv") # Reads in population data for each sample

strata(mxGenlightPopDepth) = data.frame(popData)
setPop(mxGenlightPopDepth) = ~popDepth
amova <- poppr.amova(mxGenlightPopDepth, ~popDepth) #Runs AMOVA
amova
set.seed(1999)
amovasignif <- randtest(amova, nrepet = 999) #Calculates significance levels of the AMOVA with 999 permutations
amovasignif
plot(amovasignif)

#mxGenlightPopDepth$pop=as.factor(mxGenlightPopDepth$pop)
#mxGenlightPopDepth$pop = factor(mxGenlightPopDepth$pop, levels(mxGenlightPopDepth$pop)[c(4,3,2,1,5,6,8,7)])

set.seed(694)
mx.fst <- stamppFst(mxGenlightPopDepth, nboots = 999, percent = 95, nclusters = 4) #99 permutations
mx.fst$Fsts
mx.fst$Pvalues

# #####K-Means Clustering and DAPC
# grp <- find.clusters(mxGenlightPopDepth, max.n.clust=11, glPca = pca.1, perc.pca = 100, n.iter=1e6, n.start=1000) 

pop.order <- c("Alacranes-10", "Bajos del Norte-10", "Alacranes-15", "Bajos del Norte-15", "Alacranes-25", "Bajos del Norte-25","Alacranes-35", "Bajos del Norte-35")

# reads in fst matrix
snpFstMa <- as.matrix(mx.fst$Fsts)
upperTriangle(snpFstMa, byrow=TRUE) <- lowerTriangle(snpFstMa)
snpFstMa <- snpFstMa[,pop.order] %>%
  .[pop.order,]
snpFstMa[upper.tri(snpFstMa)] <- NA
snpFstMa <- as.data.frame(snpFstMa)

snpFstMa$Pop = factor(row.names(snpFstMa), levels = unique(pop.order))

snpQMa <- as.matrix(mx.fst$Pvalues)
upperTriangle(snpQMa, byrow=TRUE) <- lowerTriangle(snpQMa)
snpQMa <- snpQMa[,pop.order] %>%
  .[pop.order,]
snpQMa[upper.tri(snpQMa)] <- NA
snpQMa <- as.data.frame(snpQMa)
snpQMa$Pop = factor(row.names(snpQMa), levels = unique(pop.order))

snpFstMa$Pop = factor(row.names(snpFstMa), levels = unique(pop.order))
snpFst = melt(snpFstMa, id.vars = "Pop", value.name = "Fst", variable.name = "Pop2", na.rm = TRUE)
snpFst = snpFst[snpFst$Pop != snpFst$Pop2,]
snpFst$Fst = round(snpFst$Fst, 3)
snpFst = snpFst %>% mutate(Fst = replace(Fst, Fst < 0, 0))
head(snpFst)

snpQ = melt(snpQMa, id.vars = "Pop", value.name = "Pval", variable.name = "Pop2", na.rm = TRUE)
snpQ = snpQ[snpQ$Pop != snpQ$Pop2,]
snpQ$Qval = p.adjust(snpQ$Pval, method = "BH")
head(snpQ)

snpHeatmapA = ggplot(data = snpFst, aes(Pop, Pop2, fill = Fst))+
  geom_tile(color = "white")+
  scale_fill_gradient2(low = "white", high = "red", midpoint = 0, limit = c(0, 0.08),
                       space = "Lab", name = expression(paste(italic("F")[ST])))+
  geom_text(data = snpFst, aes(Pop, Pop2, label = Fst), color = ifelse(snpQ$Qval <= 0.05,"black", "darkgrey"), size = ifelse(snpQ$Qval < 0.05, 6, 5), fontface = ifelse(snpQ$Qval <= 0.05, "bold", "plain")) +
  guides(fill=guide_colorbar(barwidth = 1, barheight = 12, title.position = "top", title.hjust = 0.5))+                     
  scale_y_discrete(position = "right", labels = c("Alacranes-10 m", "Bajos del Norte-10 m","Alacranes-15 m", "Bajos del Norte-15 m", "Alacranes-25 m", "Bajos del Norte-25 m","Alacranes-35 m"))+
  scale_x_discrete(labels = str_wrap(c("Bajos del Norte-10 m", "Alacranes-15 m", "Bajos del Norte-15 m", "Alacranes-25 m", "Bajos del Norte-25 m","Alacranes-35 m", "Bajos del Norte-35 m"), width = 4)) +
  #ggtitle("   SNP") +
  theme_minimal()

snpHeatmap = snpHeatmapA + theme(
  axis.text.x = element_text(vjust = 1, size = 16, hjust = 0.5, color = "black"),
  axis.text.y = element_text(size = 16, color = "black"),
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.position = "right",
  legend.direction = "vertical",
  legend.title = element_text(size = 16),
  legend.text = element_text(size = 14),
  plot.title = element_text(size = 16)
)

snpHeatmap

ggsave("./figures/figure3.tiff", plot = snpHeatmap, width = 35, height = 15, units = "cm", dpi = 300)
```

## NGSAdmix Plot
```{r, ADMIXTURE}
paletteer_d("lisa::GeorgiaOKeeffe")
colPal2=c("turquoise", "lightgoldenrod2")
#colPal2=c("#82B5C4FF", "#D3B63DFF")
mxNgsAdmix <- read.csv("mxMcavNoClones_k2k3.csv") %>% arrange(site, depthZoneM, -cluster2.2)
# mxK2$sample = factor(mxK2$sample, levels = mxK2$sample[order(-mxK2$cluster2)])
sampleCounts = plyr::count(mxNgsAdmix, c('site','depthZoneM'))
meltedList = reshape2::melt(lapply(sampleCounts$freq,function(x){c(1:x)}))

mxNgsAdmix$barPlotOrder = meltedList$value
mxNgsAdmix$depthZoneM=as.factor(mxNgsAdmix$depthZoneM)
mdat = melt(mxNgsAdmix, id.vars=c("sample", "site", "depthZoneM", "barPlotOrder"), variable.name="Ancestry", value.name="Fraction")
#mdat2$site=as.factor(mdat2$site)
mdat$depthZoneM=as.factor(mdat$depthZoneM)
levels(mdat$depthZoneM)
levels(mdat$depthZoneM) = c("10 m", "15 m", "25 m", "35 m")

p2 = ggplot(data = subset(mdat, subset = mdat$Ancestry %in% c("cluster2.1", "cluster2.2")), aes(x=barPlotOrder, y=Fraction, fill=Ancestry, order=barPlotOrder)) +
  geom_bar(stat="identity", position="fill", width=1, colour="grey25") +
  #facet_grid(.~site, scales = "free", switch = "x", space = "free") +
  facet_grid(depthZoneM ~ site, scales = "free") + #faceting plots by Depth and Site
  labs(x = "Population", y = "Ancestry") +
  ggtitle(expression(paste(italic("K"), "=2"))) +
  theme(plot.title = element_text(size=40),
        panel.grid=element_blank(),
        panel.background=element_rect(fill=NA, colour="grey25"),
        panel.spacing.x=grid:::unit(0, "lines"),
        panel.border = element_rect(fill=NA,color="black", size=2, linetype="solid"),
        axis.text.x=element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks.x=element_blank(),
        axis.ticks.y=element_blank(),
        strip.background = element_rect(fill = "white", size = 2, color="black"),
        strip.text.x = element_text(size = 30),
        strip.text.y = element_blank(),
        #strip.text=element_text(size=20, angle=90),
        legend.key=element_blank(),
        legend.position = "none",
        legend.title = element_blank()) +
  scale_x_discrete(expand=c(0, 0)) +
  scale_y_continuous(expand=c(0, 0)) +
  #scale_fill_manual(values = brewer.pal(name = "YlGnBu", n = 4), name = "Cluster") +
  scale_fill_manual(values = colPal2, name = "Cluster") +
  guides(fill=guide_legend(override.aes=list(colour=NULL)))
p2

###########
colPal3=c("turquoise", "lightgoldenrod2", "pink")
#colPal3=c("#82B5C4FF", "#D3B63DFF", "#C37E90FF")
# mxK3 <- read.csv("mxMcavNoClones_K3.csv") %>% arrange(pop, depthZoneM, -cluster2)
# # mxK3$sample = factor(mxK3$sample, levels = mxK3$sample[order(-mxK3$cluster2)])
# sampleCountsK3 = plyr::count(mxK3, c('pop','depthZoneM'))
# meltedListK3 = reshape2::melt(lapply(sampleCountsK3$freq,function(x){c(1:x)}))
# 
# mxK3$barPlotOrderK3 = meltedListK3$value
# mdat3 = melt(mxK3, id.vars=c("sample", "pop", "depthZoneM", "barPlotOrderK3"), variable.name="Ancestry", value.name="Fraction")
# #mdat2$pop=as.factor(mdat3$pop)
# mdat3$depthZoneM=as.factor(mdat3$depthZoneM)
# levels(mdat3$depthZoneM)
# levels(mdat3$depthZoneM) = c("10 m", "15 m", "25 m", "35 m")


p3 = ggplot(data = subset(mdat, subset = mdat$Ancestry %in% c("cluster3.1", "cluster3.2", "cluster3.3")), aes(x=barPlotOrder, y=Fraction, fill=Ancestry, order=barPlotOrder)) +
  geom_bar(stat="identity", position="fill", width=1, colour="grey25") +
  #facet_grid(.~pop, scales = "free", switch = "x", space = "free") +
  facet_grid(depthZoneM ~ site, scales = "free") + #faceting plots by Depth and Site
  labs(x = "Population", y = "Ancestry") +
  ggtitle(expression(paste(italic("K"), "=3"))) +
  theme(plot.title = element_text(size=40),
        panel.grid=element_blank(),
        panel.background=element_rect(fill=NA, colour="grey25"),
        panel.spacing.x=grid:::unit(0, "lines"),
        panel.border = element_rect(fill=NA,color="black", size=2, linetype="solid"),
        axis.text.x=element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks.x=element_blank(),
        axis.ticks.y=element_blank(),
        strip.background = element_rect(fill = "white", size = 2, color="black"),
        strip.text.x = element_text(size = 30),
        strip.text.y = element_text(size = 30),
        #strip.text=element_text(size=20, angle=90),
        legend.key=element_blank(),
        legend.position = "none",
        legend.title = element_blank()) +
  scale_x_discrete(expand=c(0, 0)) +
  scale_y_continuous(expand=c(0, 0)) +
  #scale_fill_manual(values = brewer.pal(name = "YlGnBu", n = 4), name = "Cluster") +
  scale_fill_manual(values = colPal3, name = "Cluster") +
  guides(fill=guide_legend(override.aes=list(colour=NULL)))
p3

combinedAdmix = (p2 | p3)
combinedAdmix

ggsave("./figures/figure4.jpg", plot = combinedAdmix, width = 70, height = 40, units = "cm", dpi = 300)
```

##ITS2-type Profiles
```{r, its2BarPlot}

its2Profs = read.csv("mxIts2ProfCounts.csv", header = TRUE, check.names = FALSE)
head(its2Profs)

its2Profs$depthZoneM = factor(its2Profs$depthZoneM, levels = c("10", "15", "25", "35"))
levels(its2Profs$depthZoneM)

its2Profs$site = factor(its2Profs$site)
its2Profs$site = factor(its2Profs$site, levels(its2Profs$site)[c(1,2)])
its2Profs = its2Profs[order(its2Profs$site, its2Profs$depthZoneM), ]
head(its2Profs)

its2Profs = its2Profs %>% arrange(site, depthZoneM,
                                  `G3b`,
                                  `G3am`,
                                  `C3/C3b`,
                                  `C3-C3fc-C40g-C3b-C3an-C21-C3fd-C3s-C3bb`,
                                  `C3an/C3-C21-C3bb`, 
                                  `C3-C21-C3gu-C3gy-C3gw-C3gx-C3gv-C3b`, 
                                  `C3-C3an-C3de-C21-C3bb-C3fc-C3b`,
                                  `C3-C3de-C3bb-C21ae-C21-C3an-C3s`, 
                                  `C3-C3fc-C21-C3b-C3an-C3fd-C3bb`, 
                                  `C3-C3de-C21-C3bb-C3an-C21ae-C65b-C3s`, 
                                  `C3-C21-C3an-C3fc-C3gz-C3b-C3bb`, 
                                  `C3-C21-C3fc-C3an-C65b-C3b-C3bb`)  
                      
                      

sampleCounts = plyr::count(its2Profs, c('site','depthZoneM'))
meltedList = reshape2::melt(lapply(sampleCounts$freq,function(x){c(1:x)}))
its2Profs$barPlotOrder = meltedList$value
its2Profs=its2Profs[c(1,ncol(its2Profs),2:(ncol(its2Profs)-1))]

colOrder2 = order(colSums(its2Profs[10:length(its2Profs[1,])]), decreasing = TRUE)+9 #Add the number of columns of metadata

its2ProfsPerc = cbind(its2Profs[,c(1:9)],its2Profs[,c(colOrder2)])
its2ProfsPerc$sum = apply(its2ProfsPerc[, c(10:length(its2ProfsPerc[1,]))], 1, function(x) {
  sum(x, na.rm = T)
})

its2ProfsPerc = cbind(its2ProfsPerc[, c(1:9)], (its2ProfsPerc[, c(10:(ncol(its2ProfsPerc)-1))] 
                                                        / its2ProfsPerc$sum))
head(its2ProfsPerc)

apply(its2ProfsPerc[, c(10:(ncol(its2ProfsPerc)))], 1, function(x) {
  sum(x, na.rm = T)
})

head(its2ProfsPerc)


gssProf = otuStack(its2ProfsPerc, count.columns = c(10:length(its2ProfsPerc[1, ])),
                       condition.columns = c(1:9)) # remove summ rows
gssProf = gssProf %>% filter(otu != "summ") %>% droplevels()

levels(gssProf$otu)

levels(gssProf$depthZoneM)
levels(gssProf$depthZoneM) = c("10 m", "15 m", "25 m", "35 m")
levels(gssProf$site)
levels(gssProf$site) = c("Alacranes", "Bajos del Norte")
levels(gssProf$depthZoneM)
levels(gssProf$site)

#gssProf %>% arrange(site, depthZoneM, count, otu)

#Construct bar plot

its2ProfsPlotA = ggplot(gssProf, aes(x = barPlotOrder, y = count, fill = factor(otu))) +
  geom_bar(position = "stack", stat = "identity", color = "black", size = 0.25, width = 1) + 
  ylab("Proportion") +
  scale_fill_paletteer_d("dichromat::Categorical_12")+
  labs(fill = expression(paste(italic("ITS2"), " type profile"))) +
  guides(fill = guide_legend(ncol = 2, reverse = FALSE)) +
  facet_grid(depthZoneM ~ site, scales = "free_x") +#faceting plots by Depth and Site
  scale_x_discrete(expand=c(0, 0)) +
  scale_y_continuous(expand=c(0, 0)) +
  theme_bw()

its2ProfsPlot = its2ProfsPlotA +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "bottom",
        legend.title = element_text(color = "black", size = 12, hjust = 0.5, angle = 90),
        legend.text = element_text(color = "black", size = 10),
        legend.key = element_blank(),
        legend.key.size = unit(0.75,"line"),
        legend.background = element_blank(),
        # panel.border = element_blank(),
        # panel.background = element_rect(fill = "white"),
        panel.grid=element_blank(),
        panel.background=element_rect(fill=NA, colour="grey25"),
        panel.spacing.x=grid:::unit(0, "lines"),
        panel.border = element_rect(fill=NA,color="black", size=1, linetype="solid"),
        plot.background = element_blank(),
        strip.text.x = element_text(size = 12),
        strip.text.y = element_text(size = 12),
        strip.background = element_rect(fill = "white", size = 1)
  )

its2ProfsPlot

ggsave("./figures/figure5.tiff", plot = its2ProfsPlot, width = 8.5, height = 8.5, unit = "in", dpi = 300)
```

##ITS2 Stats Analysis
```{r, itsStats}
set.seed(694) #setting seed allows repetition of randomized processes

its2dispS = betadisper(vegdist(its2Profs[, c(10:ncol(its2Profs))]), its2Profs$site)

anova(its2dispS)
#No significant effect of site on beta diversity

set.seed(694)

its2dispD = betadisper(vegdist(its2Profs[, c(10:ncol(its2Profs))]), its2Profs$depthZoneM)

anova(its2dispD)
#No significant effect of depth on beta diversity

#PERMANOVA
set.seed(694)
its2Adonis = adonis(its2Profs[, c(10:ncol(its2Profs))] ~ depthZoneM * site, 
                    data = its2Profs, permutations = 9999, method = "bray")

its2Adonis

set.seed(694)

its2PWAdonis = pairwise.adonis(its2Profs[, c(10:ncol(its2Profs))],
                               factors = its2Profs$depthZoneM,
                               sim.method = "bray", p.adjust.m = "BH", perm = 9999)

its2PWAdonis 
```

##ITS2 Symbiont PCoA with Bray Curtis
```{r, itsPCoA}
zooxProfs = its2Profs[,10:ncol(its2Profs)]
zooxDist=vegdist(zooxProfs, method="bray")

zooxMds = cmdscale(zooxDist, eig = TRUE, x.ret = TRUE)

# Determine percent variation captured on each axis
# Calculate the eigenvalues so later we can figure out % variation shown on each Principal Coordinate
mxZooxPcoaVar = round(zooxMds$eig/sum(zooxMds$eig)*100, 1)
mxZooxPcoaVar

# Format data to plot
mxZooxPcoaValues = zooxMds$points
mxZooxPcoaValues

zooxI2P = its2Profs[,1:9]
row.names(zooxI2P) = zooxI2P[,1]
mxZooxPcoaValues=cbind(zooxI2P, mxZooxPcoaValues)
mxZooxPcoaValues =as.data.frame(mxZooxPcoaValues, sample = rownames(mxZooxPcoaValues))
colnames(mxZooxPcoaValues)[10] <- "PCo1"
colnames(mxZooxPcoaValues)[11] <- "PCo2"
mxZooxPcoaValues$popDepth<- paste(mxZooxPcoaValues$site, "-", mxZooxPcoaValues$depthZoneM)
head(mxZooxPcoaValues)

zooxPCoA = merge(mxZooxPcoaValues, aggregate(cbind(mean.x=PCo1,mean.y=PCo2)~popDepth, mxZooxPcoaValues, mean), by="popDepth")

#snpPCoA$depthZoneM=as.factor(snpPCoA$depthZoneM)
#snpPCoA$depthZoneM = factor(snpPCoA$depthZoneM, levels(snpPCoA$depthZoneM)[c(2,1)])

# SNP PCoA biplot
mxZooxPcoaPlotA = ggplot(zooxPCoA, aes(x = PCo1, y = PCo2, color = site, fill = site, shape = depthZoneM, linetype = depthZoneM)) +
  geom_hline(yintercept = 0, color = "gray90", size = 0.5) +
  geom_vline(xintercept = 0, color = "gray90", size = 0.5) +
  #stat_ellipse(data = subset(zooxPCoA, type = "n", geom = "polygon")) + #ellipse
  scale_linetype_manual(values=c(1,2,4,3), name = "Depth Zone")+
  geom_point(aes(x = PCo1, y = PCo2, shape = depthZoneM), size = 3, alpha = 0.3, show.legend = FALSE, guide=FALSE, position=position_jitter()) + #individual's points indicated by circles
  #geom_jitter(alpha = 0.3, show.legend=FALSE)+
  scale_shape_manual(values = c(24,22,23,25), breaks=c("10","15", "25", "35"), name = "Depth Zone", labels=c("10 m","15 m", "25 m", "35 m")) +
  geom_point(aes(x = mean.x, y = mean.y, shape = depthZoneM), size = 5, color = "black") + #population centroids indicated by triangles
  #scale_fill_paletteer_d("LaCroixColoR::PassionFruit", breaks=c("Alacranes", "Bajos del Norte"), name= "Population")+
  scale_fill_manual(values = c("seagreen3", "blue"), name= "Population")+
  #scale_color_paletteer_d("LaCroixColoR::PassionFruit", breaks=c("Alacranes", "Bajos del Norte"), name= "Population", guide=FALSE) +
  scale_color_manual(values = c("seagreen3", "blue"), name= "Population")+
  xlab(paste ("PCo 1 (", mxZooxPcoaVar[1],"%)", sep = "")) + #Prints percent variation explained by first axis
  ylab(paste ("PCo 2 (", mxZooxPcoaVar[2],"%)", sep = "")) + #Prints percent variation explained by second axis
  guides(shape = guide_legend(order = 2), fill = guide_legend(override.aes = list(shape = 22, size = 4, color = NA), order = 1))+ #linetype = guide_legend(order = 3),
  theme_bw()

mxZooxPcoaPlot = mxZooxPcoaPlotA +
  theme(axis.title.x = element_text(color = "black", size = 10),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.line.x = element_blank(),
        axis.title.y = element_text(color = "black", size = 10),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.line.y = element_blank(),
        legend.position = "left",
        panel.border = element_rect(color = "black", size = 1.2),
        panel.background = element_rect(fill = "white"),
        plot.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())



mxZooxPcoaPlot

ggsave("./figures/figure6.tiff", plot = mxZooxPcoaPlot, height = 5, width = 7, units = "in", dpi = 300)
```

##Outlier Analysis
```{r, outliers}
##read in the bed file made using plink on the bcf on koko
path_to_file <- "/Users/student/Documents/GitHub/mxMcavSnp/mxMcavNoClones.bed"
mxPca <- read.pcadapt(path_to_file, type = "bed")

mxPca=read.pcadapt("/Users/student/Documents/GitHub/mxMcavSnp/mxMcavNoClones.bcf", type = "vcf")

#choose a number of K more than any number it would be

scree <- pcadapt(input = mxPca, K = 20)
plot(scree, option = "screeplot")

#Look at the scree plot and select the number K to the left of the straight line, in this case K=4

mxPcadapt <- pcadapt(mxPca, K = 4, min.maf=0) #The default minMaf is 0.05 but for some reason it still filters out SNPs (even though the dataset should already be filtered for minMaf so I set it to 0 so it wouldn't do that)
summary(mxPcadapt)
head(mxPcadapt)
head(mxPcadapt$pvalues)
qval = p.adjust(mxPcadapt$pvalues, method = "BH")
qPcadapt=as.data.frame(qval)
outsPcadapt=which(qPcadapt[,"qval"]<0.05)

source('plot_R.r')

datReefDepth = read.table("mxMcavReefDepth.baye_fst.txt",header=T)
head(datReefDepth)
table(datReefDepth[,"qval"]<0.1)
datReefDepth$locus = c(1:nrow(datReefDepth))
outsReefDepth=which(datReefDepth[,"qval"]<0.1)
plot_bayescan("mxMcavReefDepth.baye_fst.txt",FDR=0.1,add_text=F,size=0.5,highlight=outsReefDepth)
outliersReefDepth= datReefDepth %>% filter(qval < 0.1)
outsQvalReefDepth=dplyr::select(outliers, qval, locus)

datDepth = read.table("mxMcavDepth.baye_fst.txt",header=T)
head(datDepth)
table(datDepth[,"qval"]<0.1)
datDepth$locus = c(1:nrow(datDepth))
outsDepth=which(datDepth[,"qval"]<0.1)
plot_bayescan("mxMcavDepth.baye_fst.txt",FDR=0.1,add_text=F,size=0.5,highlight=outsDepth)
outliersDepth= datDepth %>% filter(qval < 0.1)
outsQvalDepth=dplyr::select(outliers, qval, locus)


## SNPs potentially under selection
genes = read.table("mcav_gene_regions.tab")
names(genes) = c("chromo","start","end","gene")

# expand gene regions ± 2000 bp
genes$start = genes$start -2000
genes$end = genes$end +2000

gnames = read.table("mcav_cog.txt", sep = "\t")
names(gnames) = c("gene", "cog", "protein")

genes = full_join(genes, gnames, by = "gene")
genes$protein=as.character(genes$protein)
genes$protein[is.na(genes$protein)]="unknown"

#how many annotated genes do we have?
nrow(genes[genes$protein!="unknown",])
head(genes)

snpLoci = read.table("mxMcavNoClones.mafs.gz", header = TRUE)
snpLoci$locus = c(1:nrow(snpLoci))

#########Annotated outliers by reef:depth
outsByGeneReefDepth = snpLoci %>% dplyr::select(locus, chromo, position) %>%
  filter(locus %in% outsReefDepth) %>%
  #filter(locus %in% outsQval) %>%
  full_join(., outsQvalReefDepth, by = "locus") %>%
  full_join(., genes, by = "chromo") %>%
  filter(position>start, position<end)

outsByGeneReefDepth = outsByGeneReefDepth[c(1:3,5:9,4)]

for(i in 1:nrow(outsByGeneReefDepth)) {
  if (outsByGeneReefDepth$qval[i] < .01) {
    outsByGeneReefDepth$sig[i] = "***"
  } else {
    if (outsByGeneReefDepth$qval[i] < .05) {
      outsByGeneReefDepth$sig[i] = "**"
    } else {
      outsByGeneReefDepth$sig[i] = "*"
    }
  }
}

outsByGeneReefDepth

#only show annotated genes by reef depth
outsAnnoReefDepth = outsByGeneReefDepth %>% filter(protein != "unknown")
outsAnnoReefDepth

df_uniqReefDepth <- unique(outsAnnoReefDepth$locus)
length(df_uniqReefDepth)
df_uniq1ReefDepth <- unique(outsAnnoReefDepth$gene)
length(df_uniq1ReefDepth)

 write.csv(x=outsAnnoReefDepth, file="annotatedOutliersReefDepth.csv")

#########Annotated outliers by depth only
outsByGeneDepth = snpLoci %>% dplyr::select(locus, chromo, position) %>%
  filter(locus %in% outsDepth) %>%
  #filter(locus %in% outsQval) %>%
  full_join(., outsQvalDepth, by = "locus") %>%
  full_join(., genes, by = "chromo") %>%
  filter(position>start, position<end)

outsByGeneDepth = outsByGeneDepth[c(1:3,5:9,4)]

for(i in 1:nrow(outsByGeneDepth)) {
  if (outsByGeneDepth$qval[i] < .01) {
    outsByGeneDepth$sig[i] = "***"
  } else {
    if (outsByGeneDepth$qval[i] < .05) {
      outsByGeneDepth$sig[i] = "**"
    } else {
      outsByGeneDepth$sig[i] = "*"
    }
  }
}

outsByGeneDepth

#only show annotated genes by reef depth
outsAnnoDepth = outsByGeneDepth %>% filter(protein != "unknown")
outsAnnoDepth


write.csv(x=outsAnnoDepth, file="annotatedOutliersDepth.csv")
```


```{r, goMWU}
mcavGo=read.table("Mcavernosa_gene2go.tab")
names(mcavGo)=c("gene", "GO")

genes = read.table("mcav_gene_regions.tab")
names(genes) = c("chromo","start","end","gene")

# expand gene regions ± 2000 bp
genes$start = genes$start -2000
genes$end = genes$end +2000

#gnames = read.table("mcav_cog.txt", sep = "\t")
#names(gnames) = c("gene", "cog", "protein")

genes = full_join(genes, mcavGo, by = "gene")
genes$GO=as.character(genes$GO)
genes$protein[is.na(genes$GO)]="unknown"

#how many annotated genes do we have?
nrow(genes[genes$protein!="unknown",])
head(genes)

snpLoci = read.table("mxMcavNoClones.mafs.gz", header = TRUE)
snpLoci$locus = c(1:nrow(snpLoci))

snpsByGene = snpLoci %>% dplyr::select(locus, chromo, position) %>%
  #filter(locus %in% highOuts) %>%
  full_join(., genes, by = "chromo") %>%
  filter(position>start, position<end, !is.na(GO))

snpsByGene$outlier = 0
for(i in 1:nrow(snpsByGene)){
  if(snpsByGene$locus[i] %in% outs){
    snpsByGene$outlier[i] = 1
    }else{
      snpsByGene$outlier[i] = 0
    }
}

snpsByGene$outlier

input=snpsByGene %>% dplyr::select(gene, outlier)

write.csv(input, file="input.csv", row.names = FALSE, quote = FALSE)
# two columns of comma-separated values: gene id, To perform standard GO enrichment analysis based on Fisher's exact test, use binary measure (0 or 1, i.e., either sgnificant or not).

# First, press command-D on mac or ctrl-shift-H in Rstudio and navigate to the directory containing scripts and input files. Then edit, mark and execute the following bits of code, one after another.


# Edit these to match your data file names: 
input="input.csv"
#input="heats.csv" # two columns of comma-separated values: gene id, continuous measure of significance. To perform standard GO enrichment analysis based on Fisher's exact test, use binary measure (0 or 1, i.e., either sgnificant or not).
goAnnotations="Mcavernosa_gene2go.tab"
#goAnnotations="amil_defog_iso2go.tab" # two-column, tab-delimited, one line per gene, multiple GO terms separated by semicolon. If you have multiple lines per gene, use nrify_GOtable.pl prior to running this script.
goDatabase="go.obo" # download from http://www.geneontology.org/GO.downloads.ontology.shtml
goDivision="MF" # either MF, or BP, or CC
source("gomwu.functions.R")


# ------------- Calculating stats
# It might take a few minutes for MF and BP. Do not rerun it if you just want to replot the data with different cutoffs, go straight to gomwuPlot. If you change any of the numeric values below, delete the files that were generated in previos runs first.

gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="/usr/bin/perl", # replace with full path to perl executable if it is not in your system's PATH already
	largest=0.1,  # a GO category will not be considered if it contains more than this fraction of the total number of genes
	smallest=5,   # a GO category should contain at least this many genes to be considered
	clusterCutHeight=0.25, # threshold for merging similar (gene-sharing) terms. See README for details.
#	Alternative="g" # by default the MWU test is two-tailed; specify "g" or "l" of you want to test for "greater" or "less" instead. 
#Module=TRUE,Alternative="g" # un-remark this if you are analyzing a SIGNED WGCNA module (values: 0 for not in module genes, kME for in-module genes). In the call to gomwuPlot below, specify absValue=0.001 (count number of "good genes" that fall into the module)
Module=TRUE # un-remark this if you are analyzing an UNSIGNED WGCNA module 
)
# do not continue if the printout shows that no GO terms pass 10% FDR.
# ----------- Plotting results
# 
# quartz()
# results=gomwuPlot(input,goAnnotations,goDivision,
#  	absValue=-log(0.05,10),  # genes with the measure value exceeding this will be counted as "good genes". This setting is for signed log-pvalues. Specify absValue=0.001 if you are doing Fisher's exact test for standard GO enrichment or analyzing a WGCNA module (all non-zero genes = "good genes").
#  #	absValue=1, # un-remark this if you are using log2-fold changes
#  	level1=0.1, # FDR threshold for plotting. Specify level1=1 to plot all GO categories containing genes exceeding the absValue.
#  	level2=0.05, # FDR cutoff to print in regular (not italic) font.
#  	level3=0.01, # FDR cutoff to print in large bold font.
#  	txtsize=1.2,    # decrease to fit more on one page, or increase (after rescaling the plot so the tree fits the text) for better "word cloud" effect
#  	treeHeight=0.5, # height of the hierarchical clustering tree
#  #	colors=c("dodgerblue2","firebrick1","skyblue2","lightcoral") # these are default colors, un-remar and change if needed
#  )
#  # manually rescale the plot so the tree matches the text 
#  # if there are too many categories displayed, try make it more stringent with level1=0.05,level2=0.01,level3=0.001.  
#  
#  # text representation of results, with actual adjusted p-values
#  results[[1]]
# 
# 
# # ------- extracting representative GOs
# 
# # this module chooses GO terms that best represent *independent* groups of significant GO terms
# 
# pcut=1e-2 # adjusted pvalue cutoff for representative GO
# hcut=0.9 # height at which cut the GO terms tree to get "independent groups". 
# 
# # plotting the GO tree with the cut level (un-remark the next two lines to plot)
# # plot(results[[2]],cex=0.6)
# # abline(h=hcut,col="red")
# 
# # cutting
# ct=cutree(results[[2]],h=hcut)
# annots=c();ci=1
# for (ci in unique(ct)) {
#   message(ci)
# 	rn=names(ct)[ct==ci]
# 	obs=grep("obsolete",rn)
# 	if(length(obs)>0) { rn=rn[-obs] }
# 	if (length(rn)==0) {next}
# 	rr=results[[1]][rn,]
# 	bestrr=rr[which(rr$pval==min(rr$pval)),]
# 	best=1
# 	if(nrow(bestrr)>1) {
# 		nns=sub(" .+","",row.names(bestrr))
# 		fr=c()
# 		for (i in 1:length(nns)) { fr=c(fr,eval(parse(text=nns[i]))) }
# 		best=which(fr==max(fr))
# 	}
# 	if (bestrr$pval[best]<=pcut) { annots=c(annots,sub("\\d+\\/\\d+ ","",row.names(bestrr)[best]))}
# }
# 
# mwus=read.table(paste("MWU",goDivision,input,sep="_"),header=T)
# bestGOs=mwus[mwus$name %in% annots,]
# bestGOs

```
### Discriminant analysis of principal components

```{r, unconstrained dapc}
dapcKPal = paletteer_d("wesanderson::Darjeeling1")[c(1,2,3)]
#flPal = paletteer_d("vapoRwave::jazzCup")[c(2:5)]

mxVcf = read.vcfR("mxMcavNoClones.bcf")
mxGenlightPopDepth = vcfR2genlight(mxVcf, n.cores = 2) # Converts the vcf file into a file format that poppr uses the "genlight" format
locNames(mxGenlightPopDepth) = paste(mxVcf@fix[,1],mxVcf@fix[,2],sep="_")
popData = read.csv("mxInds2PopsNoClones.csv") # Reads in population data for each sample

strata(mxGenlightPopDepth) = data.frame(popData)
setPop(mxGenlightPopDepth) = ~popDepth


# popData$depthZoneM = factor(popData$depthZoneM)
# popData$depthZoneM = factor(popData$depthZoneM, levels(popData$depthZoneM)[c(10,15,25,35)])



# popData$pop = factor(popData$pop)
# popData$pop = factor(popData$pop, levels(popData$pop)[c(4, 1, 3, 2)])



# popData$popDepth = factor(popData$popDepth)
# popData$popDepth = factor(popData$popDepth, levels(popData$popDepth)[c(8,7,2,1,6,5,4,3)])
# levels(popData$popdepth) = c("UK Shallow", "UK Mesophotic", "LK Shallow", "LK Mesophotic", "TB Shallow", "TB Mesophotic", "RH Shallow", "RH Mesophotic")
# 
# 

mxGenlightPopDepth$pop = factor(mxGenlightPopDepth$pop)
mxGenlightPopDepth$pop = factor(mxGenlightPopDepth$pop, levels(mxGenlightPopDepth$pop)[c(3,2,1,4,7,6,8,5)])



# Find the best number of clusters
# You may need to run without [choose.n.clust = FALSE] and [criterion = ]
# In this case you would examine the BIC plot and pick nClust interactively in the console
#n.pca is number of samples -1
#max.n.clust is the number of pops +3

# grp = find.clusters(mxGenlightPopDepth, n.pca = 96, max.n.clust = 11)
grp = find.clusters(mxGenlightPopDepth, n.pca = 96, max.n.clust = 11, choose.n.clust = FALSE, criterion = "goesup")



#3 clusters found



```



```{r, BIC plot}
#Admixture CV values
kVals1= data.frame(K = c(1:11),CV= c(0.48344, 0.47061, 0.47192, 0.47511, 0.49162, 0.51462, 0.54407, 0.57964, 0.61354, 0.64019, 0.67834))
#BIC Values
kVals2 = data.frame(K = c(1:11),BIC = grp$Kstat)

cvPlotA = ggplot(data = kVals1, aes (x = K, y = CV)) +
geom_line(color = "gray65") +
geom_point(color = "gray65", size = 1) +
geom_point(x = 2, y = 0.47061, color = "red", size = 1) +
xlab(expression(paste("Number of clusters (",italic(K),")",sep = ""))) +
ylab("Cross-Validation Error") +
coord_cartesian(ylim = c(0.47,0.68)) +
scale_x_continuous(breaks = c(seq(1,11,2))) +
scale_y_continuous(breaks = c(seq(0.47,0.68,0.02))) +
theme_bw()



cvPlot = cvPlotA +
theme(axis.title.x = element_text(color = "black", size = 8),
axis.text.x = element_text(color = "black", size = 6),
axis.ticks.x = element_line(color = "black"),
axis.title.y = element_text(color = "black", size = 8),
axis.text.y = element_text(color = "black", size = 6),
axis.ticks.y = element_line(color = "black"),
legend.position = "none",
panel.border = element_rect(color = "black", size = 1),
panel.background = element_blank(),
plot.background = element_blank(),
panel.grid.major = element_blank(),
panel.grid.minor = element_blank())

cvPlot

bicPlotA = ggplot(data = kVals2, aes (x = K, y = BIC)) +
geom_line(color = "gray65") +
geom_point(color = "gray65", size = 1) +
geom_point(x = 3, y = grp$stat, color = "red", size = 1) +
xlab(expression(paste("Number of clusters (",italic(K),")",sep = ""))) +
ylab("BIC") +
coord_cartesian(ylim = c(630,655)) +
scale_x_continuous(breaks = c(seq(1,11,2))) +
scale_y_continuous(breaks = c(seq(630,655,5))) +
theme_bw()



bicPlot = bicPlotA +
theme(axis.title.x = element_text(color = "black", size = 8),
axis.text.x = element_text(color = "black", size = 6),
axis.ticks.x = element_line(color = "black"),
axis.title.y = element_text(color = "black", size = 8),
axis.text.y = element_text(color = "black", size = 6),
axis.ticks.y = element_line(color = "black"),
legend.position = "none",
panel.border = element_rect(color = "black", size = 1),
panel.background = element_blank(),
plot.background = element_blank(),
panel.grid.major = element_blank(),
panel.grid.minor = element_blank())



bicPlot

kSelectionPlots = (cvPlot|bicPlot)

ggsave("./figures/kSelectionPlots.png", plot = kSelectionPlots, height = 4.5, width = 14, units = "in", dpi = 300)


```



```{r, unconstrained dapc plot}
table(pop(mxGenlightPopDepth), grp$grp)


#Displays frequency plot
table.value(table(pop(mxGenlightPopDepth), grp$grp))


#Figuring out the optimal number of PCs to keep
set.seed(92)
opt1 = optim.a.score(dapc(mxGenlightPopDepth, grp$grp, n.pca = 96, n.da = 3), n.sim = 99)



dapc1 = dapc(mxGenlightPopDepth, grp$grp, n.pca = opt1$best, n.da = 3)



dapc1DF = as.data.frame(dapc1$ind.coord) %>% cbind(grp = dapc1$grp)



scatter(dapc1)
scatter(dapc1,1,1, col = dapcKPal, bg = "white", scree.da = TRUE, legend = TRUE, solid = 0.8)



df1PlotA = ggplot(data = dapc1DF, aes(x = LD1, group = grp, fill = grp, color = grp)) +
geom_point(y = 0, shape = "|", size = 3, alpha = 0.7) +
geom_density(alpha = 0.7, size = 0.25) +
xlab("Discriminant function 1") +
ylab("Density") +
scale_color_manual(values = dapcKPal, name = "Cluster") +
scale_fill_manual(values = dapcKPal, name = "Cluster") +
guides(color = "none", fill = "none", shape = "none") +
theme_bw()



df1Plot = df1PlotA +
theme(axis.title.x = element_text(color = "black", size = 12),
axis.text.x = element_blank(),
axis.ticks.x = element_blank(),
axis.title.y = element_text(color = "black", size = 12),
axis.text.y = element_text(color = "black", size = 10),
axis.ticks.y = element_line(color = "black"),
legend.position = "bottom",
panel.border = element_rect(color = "black", size = 1),
panel.background = element_blank(),
plot.background = element_blank(),
panel.grid.major = element_blank(),
panel.grid.minor = element_blank())



df1Plot
```



```{r, location prior dapc}
set.seed(124)
opt2 = optim.a.score(dapc(mxGenlightPopDepth, mxGenlightPopDepth$pop, n.pca = 96, n.da = 7), n.sim = 99)


dapc2 = dapc(mxGenlightPopDepth, mxGenlightPopDepth$pop, n.pca=40, n.da = 7)
#dapc2 = dapc(mxGenlightPopDepth, mxGenlightPopDepth$pop, n.pca = opt2$best, n.da = 7)



mxDapcVar = round(dapc1$eig/sum(dapc1$eig)*100, 1)
head(mxDapcVar)

mxDapcVar2 = round(dapc2$eig/sum(dapc2$eig)*100, 1)
head(mxDapcVar2)



grpCentroids = as.data.frame(dapc1$grp.coord)
#colnames(grpCentroids) = c("popLD1", "popLD2", "popLD3", "popLD4", "popLD5", "popLD6", "popLD7")
colnames(grpCentroids) = c("popLD1", "popLD2")
grpCentroids$K = levels(grp$grp)



mxDapc = popData %>% cbind(dapc1$ind.coord) %>% cbind(K = dapc1$grp) %>% left_join(grpCentroids) 


grpCentroids2 = as.data.frame(dapc2$grp.coord)
colnames(grpCentroids2) = c("popLD1", "popLD2", "popLD3", "popLD4", "popLD5", "popLD6", "popLD7")
grpCentroids2$popDepth = levels(popData$popDepth)
mxDapc2 = popData %>% cbind(dapc2$ind.coord) %>% left_join(grpCentroids2) %>% cbind(K = dapc1$grp)

```



```{r, dapc plot by cluster}

dapcKPal = paletteer_d("wesanderson::Darjeeling1")[c(1,2,3)]
mxDapc$depthZoneM=as.factor(mxDapc$depthZoneM)
mxDapcKPlotA = ggplot(mxDapc, aes(x = LD1, y = LD2, fill = K, shape = depthZoneM)) +
geom_hline(yintercept = 0, color = "gray90", size = 0.5) +
geom_vline(xintercept = 0, color = "gray90", size = 0.5) +
geom_segment(aes(x = popLD1, y = popLD2, xend = LD1, yend = LD2, color = K),alpha = 0.5) +
geom_point(aes(x = LD1, y = LD2, shape = depthZoneM, color = K), size = 2, alpha = 0.5, show.legend = TRUE) +
scale_shape_manual(values = c(24,22,23,25), breaks=c("10","15", "25", "35"), name = "Depth Zone", labels=c("10 m","15 m", "25 m", "35 m")) +
# stat_ellipse(data = sintDapc, type = "t" , geom = "polygon", alpha = 0.1, fill = NA, size = 0.4) +
#geom_point(aes(x = popLD1, y = popLD2, shape = depthZoneM), size = 4, color = "gray15") +
# geom_label(aes(x = popLD1, y = popLD2, label = popdepth), fill = "white", size = 2) +
scale_fill_manual(values = dapcKPal, name = "Cluster") +
scale_color_manual(values = dapcKPal, guide = NULL) +
xlab(paste("DA 1 (", mxDapcVar[1], "%)", sep = "")) +
ylab(paste("DA 2 (", mxDapcVar[2], "%)", sep = "")) +
guides(fill = guide_legend(override.aes = list(shape = 22, color = NA, alpha = NA), order = 1)) +
theme_bw()



mxDapcKPlot = mxDapcKPlotA +
theme(axis.title.x = element_text(color = "black", size = 10),
axis.text.x = element_blank(),
axis.ticks.x = element_blank(),
axis.line.x = element_blank(),
axis.title.y = element_text(color = "black", size = 10),
axis.text.y = element_blank(),
axis.ticks.y = element_blank(),
axis.line.y = element_blank(),
legend.position = "right",
panel.border = element_rect(color = "black", size = 1),
panel.background = element_rect(fill = "white"),
panel.grid.major = element_blank(),
panel.grid.minor = element_blank())



mxDapcKPlot

ggsave("./figures/clusterDapc.png", plot = mxDapcKPlot, height = 4.5, width = 7, units = "in", dpi = 300)

```



```{r, locprior K dapc plot}
mxPal=c("seagreen3", "blue")
mxDapc2$depthZoneM=as.factor(mxDapc$depthZoneM)
mxDapcPlotA = ggplot(mxDapc2, aes(x = LD1, y = LD2, fill = site, shape = depthZoneM)) +
geom_hline(yintercept = 0, color = "gray90", size = 0.5) +
geom_vline(xintercept = 0, color = "gray90", size = 0.5) +
geom_segment(aes(x = popLD1, y = popLD2, xend = LD1, yend = LD2, color = site),alpha = 0.5) +
geom_point(aes(x = LD1, y = LD2, shape = depthZoneM, color = site), size = 2, alpha = 0.5, show.legend = TRUE) +
scale_shape_manual(values = c(24,22,23,25), breaks=c("10","15", "25", "35"), name = "Depth Zone", labels=c("10 m","15 m", "25 m", "35 m")) +
# stat_ellipse(data = sintDapc, type = "t" , geom = "polygon", alpha = 0.1, fill = NA, size = 0.4) +
geom_point(aes(x = popLD1, y = popLD2, shape = depthZoneM), size = 4, color = "gray15") +
# geom_label(aes(x = popLD1, y = popLD2, label = popdepth), fill = "white", size = 2) +
scale_fill_manual(values = mxPal, name = "Reef") +
scale_color_manual(values = mxPal, guide = NULL) +
xlab(paste("DA 1 (", mxDapcVar2[1], "%)", sep = "")) +
ylab(paste("DA 2 (", mxDapcVar2[2], "%)", sep = "")) +
guides(fill = guide_legend(override.aes = list(shape = 22, color = NA, alpha = NA), order = 1)) +
theme_bw()



mxDapcPlot = mxDapcPlotA +
theme(axis.title.x = element_text(color = "black", size = 10),
axis.text.x = element_blank(),
axis.ticks.x = element_blank(),
axis.line.x = element_blank(),
axis.title.y = element_text(color = "black", size = 10),
axis.text.y = element_blank(),
axis.ticks.y = element_blank(),
axis.line.y = element_blank(),
legend.position = "right",
panel.border = element_rect(color = "black", size = 1),
panel.background = element_rect(fill = "white"),
panel.grid.major = element_blank(),
panel.grid.minor = element_blank())



mxDapcPlot

ggsave("./figures/locDapc.png", plot = mxDapcPlot, height = 4.5, width = 7, units = "in", dpi = 300)

```


